app/controllers/blogs_controller.rb

```diff
@@ -2,14 +2,16 @@
 
 class BlogsController < ApplicationController
   skip_before_action :authenticate_user!, only: %i[index show]
-
-  before_action :set_blog, only: %i[show edit update destroy]
+  before_action :set_own_blog, only: %i[edit update destroy]
 
   def index
     @blogs = Blog.search(params[:term]).published.default_order
   end
 
-  def show; end
+  def show
+    blog = Blog.find(params[:id])
+    @blog = user_signed_in? && blog.user_id == current_user.id ? blog : Blog.published.find(params[:id])
```

Comment:
```suggestion
    @blog = Blog.some_scope.find(...)
```

のようにscopeを使ってSQL一発で取ってきた方が効率が良い＆再利用もできますー。

Reply:
scopeで一発で取ってくる方法が中々思いつかなったので、ご教授願いたいです🙇🏻‍♂️